test-data/unit/check-python310.test

```diff
@@ -1484,6 +1484,60 @@ def g(m: Medal) -> int:
             reveal_type(m)  # N: Revealed type is "Literal[__main__.Medal.bronze]"
             return 2
 
+
+[case testMatchLiteralPatternEnumWithProperty]
+from enum import Enum
+from typing import NoReturn
+def assert_never(x: NoReturn) -> None: ...
+
+class int:
+    def __new__(cls, value: int): pass
+
+class Medal(int, Enum):
+    prize: str
```

Comment:
yup, should be able to tomorrow

Reply:
I thought about this a bit, and this is either a request for a `nonmember(...)` test, which is currently not supported, or an enum like the one above. I've added a test for `nonmember` that's been marked as expected to fail: https://github.com/python/mypy/pull/17207/files#diff-4f8d4b202949b4cf2221e3295fbbd1774e152e44b141880981895c349599e927R1-R36

Let me know if that's what you wanted, and/or if we should keep that

Reply:
A class attribute that's not declared as `nonmember` is an enum member is it not? The case below is a different type, but seems like what you were suggesting if you didn't mean `nonmember`

```python
class Medal(Enum):
    gold = 1
    silver = 2
    bronze = 3
```

Or did you want one that declared a type? I don't remember if I saw one with a type, but if that's all you meant I can add that. Otherwise I'm going to need some guidance to what you're requesting.

Reply:
Or did you want one that also implements `__new__`?

Reply:
I specifically want an *annotated* attribute. The specification says that should be a type checker error: https://typing.readthedocs.io/en/latest/spec/enums.html#defining-members

Reply:
I'm sorry, I was even thinking that seems odd to put typing on an attribute, but I guess I didn't realize you wanted a failing test case (and I didn't realize it was in the spec, I just didn't think people would want to do that for probably the very reason it's declared :sweat_smile: )

Reply:
I was looking at adding the test and was also looking to see if anything matched that test case already (just in case it did). I came across https://github.com/python/mypy/blob/740d39ecc31c64675439e6796dcf46e6a9110896/test-data/unit/check-enum.test#L1755-L1781 which is actually a test case that was written to support a change that fixed an issue _I_ reported. Go figure!

It seems like this is currently not a type checker error and I can look at possibly adding it, but I will have to look either over the weekend or next week. I can't really write an xfail test since I'm not really sure what the error message would be (I guess it wouldn't match a not failed test so the error message might not be that important).

Just calling out that I did look into this, but will not be able to complete it immediately

Reply:
I think what Jelle is asking for is something like this (you'll also need to revert the changes to check-python311.test):
```patch
diff --git a/mypy/semanal.py b/mypy/semanal.py
index d2f02d483..7f23904e6 100644
--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -4200,6 +4200,8 @@ class SemanticAnalyzer(
                     ),
                     lvalue,
                 )
+            if explicit_type and has_explicit_value:
+                self.fail("Type annotations are not allowed for enum members", lvalue)

         if (not existing or isinstance(existing.node, PlaceholderNode)) and not outer:
             # Define new variable.
diff --git a/test-data/unit/check-enum.test b/test-data/unit/check-enum.test
index d53935085..0f41865ad 100644
--- a/test-data/unit/check-enum.test
+++ b/test-data/unit/check-enum.test
@@ -1764,7 +1764,7 @@ class B(A):
     x = 1  # E: Cannot override writable attribute "x" with a final one

 class A1(Enum):
-    x: int = 1
+    x: int = 1  # E: Type annotations are not allowed for enum members
 class B1(A1):  # E: Cannot extend enum with existing members: "A1"
     pass

@@ -2185,3 +2185,12 @@ reveal_type(A.y.value)  # N: Revealed type is "Literal[2]?"
 def some_a(a: A):
     reveal_type(a.value)  # N: Revealed type is "Union[Literal[1]?, Literal[2]?]"
 [builtins fixtures/dict.pyi]
+
+[case testErrorOnAnnotatedMember]
+from enum import Enum
+
+class Medal(Enum):
+    gold: int = 1  # E: Type annotations are not allowed for enum members
+    silver: str = 2  # E: Type annotations are not allowed for enum members \
+                     # E: Incompatible types in assignment (expression has type "int", variable has type "str")
+    bronze = 3
```

Reply:
Thanks! I primarily want a test case to make sure this case is covered by the tests. Ideally of course we should follow the spec and raise an error, but that can wait for a later PR too.

Reply:
Thanks @hamdanal, I finally understood the request, but I'll check out the implementation you provided! I was just headed to sleep and didn't know if I'd have time this weekend to address the issue so I was just following up! I'll try to get to this in the next couple days

Reply:
updated